e2e docs (#464)
vtex/faststore: docs/e2e-testing.md

```json
{
  "baseUrl": "http://localhost:8000",
  "integrationFolder": "cypress/e2e"
}
```

5. Now, run `npm run test:e2e` to [run tests in the command line](https://docs.cypress.io/guides/guides/command-line.html#Installation) or `npm run cy:open` to open the [Cypress dashboard](https://www.cypress.io/dashboard). Whatever option you choose will generate a folder structure as below:

   ```bash
   ├── cypress
   │   ├── fixtures
   │   	├── example.json
   │	├── integration
   │   	├── examples
   │   		├── ...
   │   ├── support
   │   ├── plugins
   ```

Now you're ready to start developing your own E2E tests! Additionally, you can install the VTEX helper functions hub by running (command - WIP) to make writing E2E tests easier.

### Step 2: Writing a sample test

In the following, we'll create a simple test that verifies how a specific product search behaves. Use it as an example to create your own E2E test with Cypress.

1. Open up your SFJ project in the code editor of your choosing.
2. Create a new file named `index.js` inside `cypress/e2e/{test-name}` (e.g., `cypress/e2e/search`).
3. Using JavaScript, write your E2E test. Take the code below as an example:

>    ```bash

Reviewer: ```suggestion
   ```js
```

## Overview

> [Cypress](https://github.com/cypress-io/cypress) is a testing framework that provide a great experience to write end-to-end (E2E) tests for modern web applications. It's very flexible, fast, easy to use, and besides being broadly used in the javascript ecosystem, it works very well with Gatsby, that's why we choose it as our main tool.

Reviewer: ```suggestion
[Cypress](https://github.com/cypress-io/cypress) is a testing framework that provides a great experience to write end-to-end (E2E) tests for modern web applications. It's very flexible, fast, easy to use, and besides being broadly used in the JavaScript ecosystem, it works very well with Gatsby, that's why we choose it as our main tool.
```

> Feel free to check some other alternatives and compare the trade-offs of end-to-end test tools [here](https://stackshare.io/cypress/alternatives).

Reviewer: Do you think that is necessary?

Author: No, we can remove it!

## Getting Started

1. Install the `cypress` and the `start-server-and-test`(permit using Gatsby’s development server with Cypress).

   ```bash
     $ yarn add cypress start-server-and-test --dev
   ```

> 2. Add theses scripts into your `package.json`. `cy:open` will open the cypress [dashboard](https://www.cypress.io/dashboard) and `test:e2e` will run all the tests in the command line.

Author: Can we keep the explanations somewhere?

> 2. Add theses scripts into your `package.json`. `cy:open` will open the cypress [dashboard](https://www.cypress.io/dashboard) and `test:e2e` will run all the tests in the command line.

Reviewer 2: @vitorflg Coloquei depois do código como um callout. Você acha melhor colocar antes?

       "cy:open": "cypress open"
     }
   }
   ```

3. Update the `cypress.json` config file.

```json
{
  "baseUrl": "http://localhost:8000",
  "integrationFolder": "cypress/e2e"
}
```

4. Run `npm run test:e2e`(it runs tests [in the command line](https://docs.cypress.io/guides/guides/command-line.html#Installation)) or `npm run cy:open`(it opens the [cypress dashboard](https://www.cypress.io/dashboard)). Both commands will generate a folder structure as below:
   ```bash
   ├── cypress
   │   ├── fixtures
   │   	├── example.json
   │	├── integration
   │   	├── examples
   │   		├── ...
   │   ├── support
   │   ├── plugins
   ```

## Writing a sample test

Let's create a simple test that cover a specific product search.

1. Create a file at `cypress/e2e/search/index.js`.
2. Paste the below code.

   ```bash
   describe("Search e2e tests" () => {

   	beforeEach(() => {
   	  cy.visit("/")
   	})

   	it("Finds a very specific product using the search bar", () => {
   		cy.get("input[aria-label="Search"]").type("Eyeglasses")

   		cy.contains('li', '[PRODUCT_NAME]').click()

   		cy.contains('h1', '[PRODUCT_NAME')
   	})
   })

   ```

> 

Reviewer 2: ```suggestion

>ℹ️ *This test checks how our SFJ store behaves when it receives "Eyeglasses" as an input in our search bar.*

>ℹ️ *To build your own tests, we strongly encourage you to check Cypress's [guides](https://docs.cypress.io/guides/getting-started/writing-your-first-test.html#Add-a-test-file), [examples](https://docs.cypress.io/examples/examples/recipes.html#Fundamentals), [API reference](https://docs.cypress.io/api/api/table-of-contents.html), and [best practices](https://docs.cypress.io/guides/references/best-practices.html).*

```

> 3. Run `npm run test:e2e` and verify if all tests are passing.

Reviewer 2: ```suggestion
4. Run `npm run test:e2e` to check the status of the tests.
```

> ## Code Coverage
> 
> To visualize the project code coverage just run `npm run coverage`... (WIP)
> 
> ## Hooks
> 
> VTEX has a hub of helper functions to make writing E2E tests easy, you can install .. (WIP)

Reviewer 2: ```suggestion

That's it! Additionally, you can run `npm run coverage` (WIP) to visualize the test's code coverage.

```

> ## CI/CD Integration
> 
> WIP...
> 
> ## References
> 
> [Cypress Documentation](https://docs.cypress.io/guides/overview/why-cypress.html#In-a-nutshell)
> 
> [Cypress API](https://docs.cypress.io/api/api/table-of-contents.html)
> 
> [Cypress Examples](https://docs.cypress.io/examples/examples/recipes.html)
> 
> [VTEX Examples](https://github.com/vtex-sites/storecomponents.store)
> 
> [Best Practices](https://docs.cypress.io/guides/references/best-practices.html)

Reviewer 2: ```suggestion
```
